.github/workflows/continuous-integration.yml

```diff
@@ -22,6 +22,16 @@ jobs:
       - name: "Validate composer.json and composer.lock"
         run: php7.2 $(which composer) validate --strict
 
+      - name: "Restore composer dependencies"
+        uses: actions/cache@preview
+        id: cache
+        with:
+          path: ~/.composer/cache/files
+          key: composer-cache-${{ github.ref }}
```

Comment:
`${{ github.ref }}` needs to be hashed somehow as github actions is throwing a 404 when saving the cache, as it has slashes in it.

Reply:
this key needs to be unique here, so newer caches are persisted, otherwise we get:
`Cache hit occurred on the primary key `.
Somethine like:
`key: composer-cache-${{ github.ref }}-${{ job.id }}`
But Github does not provide the job number or anything unique

Reply:
using `github.sha` for uniqueness now.